tests/src/test/scala/org/http4s/EntityDecoderSuite.scala

```diff
@@ -330,17 +330,16 @@ class EntityDecoderSuite extends Http4sSuite {
 
   val request = Request[IO]().withEntity("whatever")
 
-  test("apply should invoke the function with  the right on a success") {
+  test("apply should invoke the function with the right on a success") {
     val happyDecoder: EntityDecoder[IO, String] =
       EntityDecoder.decodeBy(MediaRange.`*/*`)(_ => DecodeResult.success(IO.pure("hooray")))
-    IO.async_[String] { cb =>
+    IO.async[String] { cb =>
       request
         .decodeWith(happyDecoder, strict = false) { s =>
           cb(Right(s))
           IO.pure(Response())
         }
-        .unsafeRunSync()
-      ()
+        .as(Some(IO.unit))
```

Comment:
shouldn't this return `None`? `async` has an optional cancelation token, and this is not really canceling anything right?

Reply:
👍 
yes you're right.

Reply:
https://github.com/http4s/http4s/pull/4194/commits/53e61553a68eb68aff759388503de3ba9459344f